numpy/core/numeric.pyi

```diff
@@ -51,6 +55,15 @@ _ArrayType = TypeVar("_ArrayType", bound=NDArray[Any])
 
 _CorrelateMode = Literal["valid", "same", "full"]
 
+class _Unknown: ...
+
+_ArrayLikeUnknown = Union[
+    _SupportsArray["dtype[_Unknown]"],
+    _NestedSequence[_SupportsArray["dtype[_Unknown]"]],
+    _Unknown,
+    _NestedSequence[_Unknown],
+]
+
```

Comment:
Right, 3 things hre:
* Can you move this to the other ArrayLike type aliases over in [numpy/typing/_array_like](https://github.com/numpy/numpy/blob/main/numpy/_typing/_array_like.py)?
* I'd suggest the name `_UnknownType` rather than `_Unknown`
* Could you add a small comment explaining its purporse, maybe in combination with a link to this PR for future reference?